src/SFA.DAS.EAS.Portal/Application/Commands/Cohort/CohortApprovalRequestedCommandHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using SFA.DAS.Commitments.Api.Client.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using System;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.Types;
+
+namespace SFA.DAS.EAS.Portal.Application.Commands.Cohort
+{
+    public class CohortApprovalRequestedCommandHandler : ICommandHandler<CohortApprovalRequestedCommand>
+    {
+        private readonly IAccountDocumentService _accountsService;
+        private readonly IProviderCommitmentsApi _providerCommitmentsApi;
+
+        public CohortApprovalRequestedCommandHandler(
+            IAccountDocumentService accountsService, 
+            IProviderCommitmentsApi providerCommitmentsApi)
+        {
+            _accountsService = accountsService;
+            _providerCommitmentsApi = providerCommitmentsApi;
+        }
+
+        public async Task Handle(CohortApprovalRequestedCommand command, CancellationToken cancellationToken = default)
+        {
+            var accountDocumentTask = _accountsService.Get(command.AccountId);
+            var commitmentTask = _providerCommitmentsApi.GetProviderCommitment(command.ProviderId, command.CommitmentId);
+            await Task.WhenAll(accountDocumentTask, commitmentTask);
```

Comment:
No need to use `WhenAll`, we await the tasks below.

Reply:
Will run in parallel with WhenAll.  The awaits below are just to unwrap any exceptions.

Reply:
The tasks will still run concurrently without the `WhenAll`. Try it! 😄

Reply:
An await suspends the remaining code and returns the rest of the code as a continuation to the caller.   So multiple awaits run sequentially whereas WhenAll will run the tasks in parallel.  If the tasks have no interdependence then a performance benefit to use a WhenAll.

Reply:
Async tasks (when not immediately awaited) don't block the current thread, even if you don't use `WhenAll`.

It's actually more efficient _not_ to use `WhenAll` (although only marginally in this case). `WhenAll` completes when all the tasks have completed. `await` completes when the awaited task completes, which leaves a window between awaiting the first task and awaiting the second task, that allows work to be done whilst the second task is running, which otherwise would be spent waiting for the second task to complete. As a bonus, less code means a smaller memory footprint, which potentially means less cache misses, boosting performance.

Reply:
`WhenAll` removed 👍.